synapse/handlers/register.py

```diff
@@ -399,14 +399,14 @@ def get_or_create_user(self, localpart, displayname, duration_seconds):
 
             yield registered_user(self.distributor, user)
         else:
-            yield self.store.flush_user(user_id=user_id)
+            yield self.store.user_delete_access_tokens(user_id=user_id)
```

Comment:
We specifically want to invalidate old tokens when we re-register? If so it's probably worth updating the docstring as:

> Creates a new user or returns an access token for an existing one

implies existing tokens are preserved.